### DISCUSSION TITLE
How to read local files in Nuxt 3...

### DISCUSSION THREAD

**Discussion by GBKS** [nuxt/nuxt#17806] [Questions]
I am trying to replicate this [markdown blog tutorial](https://regenrek.com/posts/create-a-frontmatter-markdown-powered-blog-with-nuxt.js/#5-list-all-your-blogposts) with Nuxt 3, but stumbling on the part of reading a folder of markdown files and passing it into the page.

The issues are as follows:
- `asyncData` is no longer available. I addressed this (I think) by using `setup()`
- `require is not defined`, I am not sure why. Might be related to using a new node version ([see here](https://stackoverflow.com/questions/31931614/require-is-not-defined-node-js)). None of the fixesI found worked for me
- Tried to use `fs.readdir` instead, but it tells me that it can't find the `fs` module

Not sure what I am missing as it's a fairly straightforward task. But after looking through 50+ pages of StackOverflow questions, Github issues, etc, I am not making progress. All I want to do is get a list of markdown files from a folder, on the server, and pass it to the page to list out. Thanks in advance for your help.

**Comment by wobsoriano** [comment]
I think if you want to make this work you have to disable vite

```ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    vite: false
})
```

then

```html
<script setup lang="ts">
if (process.server) {
    const fs = await import('fs')
    // do something with fs
}
</script>
```

or add a markdown loader for webpack to read markdown files (if you disabled vite)

also worth mentioning https://github.com/antfu/vite-plugin-md

**Comment by hermesalvesbr** [comment]
I'm trying to work with local files too.

```

import fse from "fs-extra";
async function copyFiles() {
  try {
    await fse.copy(PATH_TO_PDF_FILE, "assets/pdf/tmp/mynewfile.pdf");
    console.log("success!");
  } catch (err) {
    console.error(err);
  }
}
```

Always show me this error message: 

polyfills.js:6 Uncaught TypeError: Cannot read properties of undefined (reading 'GRACEFUL_FS_PLATFORM')

![Screenshot_20220323_152745](https://user-images.githubusercontent.com/379850/159770433-9063e9d3-ccc8-4552-9a86-afc89bad054d.png)

**Any idea to make work this code?** 

Thanks for helping ;-)

**Reply by OverGreed** [reply]
I have also one question does it also should work layouts?
